network.go

```diff
@@ -944,7 +945,10 @@ func (n *network) ipamAllocateVersion(ipVer int, ipam ipamapi.Ipam) error {
 		// irrespective of whether ipam driver returned a gateway already.
 		// If none of the above is true, libnetwork will allocate one.
 		if cfg.Gateway != "" || d.Gateway == nil {
-			if d.Gateway, _, err = ipam.RequestAddress(d.PoolID, net.ParseIP(cfg.Gateway), nil); err != nil {
+			var gatewayOpts = map[string]string{
+				netlabel.Gateway: "",
```

Comment:
@rmb938 i was thinking more like adding a new constant/label in ipamapi package `RequestAddressType` as the key and the value can be `netlabel.Gateway` :)